apps/publikator/components/editor/modules/memo/MemoTree.js

```diff
@@ -1,116 +1,178 @@
-import React, { useState } from 'react'
+import React, { useMemo, useState, useContext } from 'react'
 import { compose } from 'react-apollo'
 
 import {
+  UnpublishIcon,
+  EditIcon,
   DiscussionContext,
+  CommentNode,
   CommentComposer,
-  CommentComposerPlaceholder,
-  CommentList
+  CommentComposerPlaceholder
 } from '@project-r/styleguide'
 
 import withT from '../../../../lib/withT'
 import withMe from '../../../../lib/withMe'
+
 import { withMemos, getDisplayAuthor } from './graphql'
 
-const buildDiscussionContext = ({
-  me,
-  repoId,
-  onPublished,
-  onEdited,
-  onUnpublished,
-  publish,
-  edit,
-  unpublish,
-  t
-}) => ({
-  discussion: {
-    id: repoId,
-    displayAuthor: getDisplayAuthor(me),
-    rules: {
-      maxLength: null,
-      minInterval: null,
-      disableTopLevelComments: false,
-      anonymity: 'FORBIDDEN'
+const Tree = props => {
+  const {
+    repoId,
+    parentId,
+    onPublished,
+    publish,
+    edit,
+    unpublish,
+    memos,
+    me,
+    t
+  } = props
+
+  const discussionContext = useMemo(() => {
+    return {
+      t,
+      discussion: {
+        id: repoId,
+        displayAuthor: getDisplayAuthor(me)
+      },
+      actions: {
+        submitMemo: (parentId, text) =>
+          publish(repoId, parentId, text).then(
+            async response => {
+              !!onPublished && (await onPublished(response.data.memo))
+              return { ok: true }
+            },
+            error => ({ error })
+          ),
+        editMemo: (id, text) =>
+          edit(id, text).then(
+            () => ({ ok: true }),
```

Comment:
do we use this `{ ok: true }` object somewhere?

Reply:
For submit and edit mutation, [yes](https://github.com/republik/plattform/blob/publikator-refactor-memos/packages/styleguide/src/components/Discussion/Composer/CommentComposer.js#L247-L248): `CommentComposer` calls handler (`onSubmit`) and then processes response – ok or error – to handle post composer matters.

For unpublish, no. But kept it, for sanity purposes, the same.